src/main/flight/pid.c

```diff
@@ -740,6 +741,48 @@ float pidGetAirmodeThrottleOffset(void)
 }
 #endif
 
+// when new Rx data is received, update the maxDeflectionAbs value, so we don't do this every PID loop
+static float maxDeflectionAbs = 0.0f;
+float sendMaxDeflectionAbs(float maxRcDeflectionAbs)
```

Comment:
There are several problems here.
1) Function naming: the intent is not sending something, it is to get the value of a variable, so `getXYZ` is better suited.
2) Efficiency: This is calculating the absolute number on every function call. If you were to call this function multiple times in a loop (for example because you need this value in multiple places) the software would do many unneeded calculations. Better to just update a static variable once per loop and then just return the value when calling the function.
3) Placement: This function definition belongs into the `rc.c` file

Reply:
The intent was that this function would run in pid.c *only* when rc.c had new data.  It's run when rc.c's `processRcCommand` gets true on `isRxDataNew`, and not run at any other time.   So in a sense the value is being 'sent' by rc.c, when rc.c has new data.  That's why I have it the strange name, it's not a 'get' function in pid.c.  So it runs infrequently, but when it does it updates the static value of `maxDeflectionAbs` in pid.c.

You know more than I do of these things, I thought that the function `sendMaxDeflectionAbs` was defined in rc.h, run in rc.c, and then in pid.c had to be instantiated again in order to work?

Reply:
These (inverse) dependencies are usually problematic. And it gets really ugly when mutexes are used (which may be BF case in future). In a lot of cases, it is easier to recalculate the value.

Quite clean implementation is to queue event from RC to PID. PID can then process received events at desired time, acting accordingly. But no such mechanism is available yet.

In this case, it may be enough to call `pidNotify_RC()` after RC values are updated and recalculate the value there (using getMaxRcDeflectionAbs() call to get values from RC).

Reply:
So... in `rc.c` we make a static boolean for `pidNotifyNewRcData` or something, and set this true in `rc.c / processRcCommand()` on receipt of new data.
And the relevant EzLanding code then sits inside a check for `if (pidNotifyNewRcData ()`, which checks every loop for true.  If true when run from PIDs, they do the EzLanding stuff, and in rc.c where it was called, it gets set to false, waiting a transition to true?
Seems complex but would work.  I'll try.

Reply:
I'll try a solution of that kind.  Thanks.

Reply:
@ctzsnooze : It is possible, but complex.

The idea is to make public (callback) function `pidNotify_RC` (or `pidNotify_onRcEvent`, `pidNotify_onRcChange` etc.)  in pid.c, that will update PID state after called by RC. Actual PID/ezLanding function will use internal PID-related state (maxDeflectionAbs, but there may be better name). Eventually, there may be more centralized/generic mechanism to handle this, but it will be easy to refactor.

In `rc.c`, just call `pidNotify_RC` when RC data change (in processRcCommand). 

The idea is that RC code only knows that PID wants to be notified, but doesn't care about details. And PID/ezLanding code updates self when necessary, using public RC function, but not caring much about RC internals.